### PR TITLE
:arrow_up::one::two: apps_odoo_com

### PR DESCRIPTION
**Pull request by fedoranvar**

### CODE REVIEW

**Comment by fedoranvar** [comment]
image

File: apps_odoo_com/doc/index.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+.. Known issues / Roadmap
+.. ======================
+..
+.. * ...
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
Comment (yelizariev) [inline apps_odoo_com/doc/index.rst]: what

File: apps_odoo_com/models/stats.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 # License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
-from openerp import models, fields
+from odoo import models, fields
 
 
 class User(models.Model):
 
     _name = 'apps_odoo_com.user'
     _order = 'odoo_id desc'
+    _description = 'description of user'
```
Comment (yelizariev) [inline apps_odoo_com/models/stats.py]: _description = 'Apps Store User'

File: apps_odoo_com/xmlrpc.py [hunk 3]
```diff
@@ -1,28 +1,32 @@
...
-    common = xmlrpclib.ServerProxy('{}/xmlrpc/2/common'.format(url))
-    models = xmlrpclib.ServerProxy('{}/xmlrpc/2/object'.format(url))
+    common = xmlrpc.client.ServerProxy('{}/xmlrpc/2/common'.format(url))
+    models = xmlrpc.client.ServerProxy('{}/xmlrpc/2/object'.format(url))
     uid = common.authenticate(dbname, username, password, {})
     if not uid:
         # for security reason log password only in debug level
         _logger.debug('Authenticate failed: %s', (dbname, username, password))
-        raise UserError(_('Authentication failed: %s'), (dbname, username,))
+        raise UserError(_('Authentication failed: %s'))
```
Comment (yelizariev) [inline apps_odoo_com/xmlrpc.py]: It will not work. Why did you delete vars?

Comment (fedoranvar) [reply apps_odoo_com/xmlrpc.py]: It was throwing an error:

TypeError: Takes 2 positional arguments, but receive 3

File: apps_odoo_com/doc/index.rst (hunk 1 above)
Comment (fedoranvar) [reply apps_odoo_com/doc/index.rst]: Left when I was porting file-structure of module from 8 to 12

File: apps_odoo_com/security/res_groups.xml [hunk 4]
```diff
@@ -3,26 +3,22 @@
  Copyright 2017 IT-Projects LLC (<https://it-projects.info>)
  License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
 -->
-<openerp>
-<data>
+<odoo>
 
+  <record id="category" model="ir.module.category">
```
Comment (KolushovAlexandr) [inline apps_odoo_com/security/res_groups.xml]: keep commit history clean

File: apps_odoo_com/data/ir_config_parameter_data.xml [hunk 5]
```diff
@@ -3,20 +3,18 @@
...
-<openerp>
-<data noupdate="1">
+<odoo noupdate="1">
 
-<record id="param_apps_odoo_com_login" model="ir.config_parameter">
-    <field name="key">apps_odoo_com.login</field>
-    <field name="value"> </field>
-    <field name="group_ids" eval="[(4, ref('apps_odoo_com.apps_odoo_com_group_manager'))]"/>
-</record>
+  <record id="param_apps_odoo_com_login" model="ir.config_parameter">
```
Comment (KolushovAlexandr) [inline apps_odoo_com/data/ir_config_parameter_data.xml]: keep commit history clean

File: apps_odoo_com/models/res_config.py [hunk 6]
```diff
@@ -13,21 +12,22 @@ class Settings(models.TransientModel):
 
     _name = 'apps_odoo_com.settings'
```
Comment (KolushovAlexandr) [inline apps_odoo_com/models/res_config.py]: integrate it into res.config.settings

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]

File: apps_odoo_com/doc/index.rst [hunk 7]
```diff
@@ -0,0 +1,59 @@
+==================================
```
Comment (KolushovAlexandr) [inline apps_odoo_com/doc/index.rst]: check indents

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
check lints, runbot warnings and you broke res.config.settings model. as I said try to save and load settings  not only for this module configs but for other modules too

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
res.config.settings has its own common view use it instead of creating a new one. And again, check runbot warnings

File: apps_odoo_com/models/res_config.py [hunk 8]
```diff
@@ -1,33 +1,33 @@
...
-        self.ensure_one()
-
-        for field_name, key_name in PARAMS:
-            value = getattr(self, field_name, '').strip()
-            self.env['ir.config_parameter'].set_param(key_name, value)
+    def set_values(self):
+        super(ResConfigSettings, self).set_values()
+        config_parameters = self.env["ir.config_parameter"].sudo()
+        for record in self:
+            config_parameters.sudo().set_param("apps_odoo_com.login", record.apps_login)
```
Comment (KolushovAlexandr) [inline apps_odoo_com/models/res_config.py]: sudo() is already applied on 18th line

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
* runbot hav not passed 
``
WARNING | server | odoo.models:3865 _load_records Creating record __apps_odoo_com__.user_145731 in module apps_odoo_com.
-- | -- | --
``
* there are some default values in the res.config model fields it's ' '. Get rid of them

File: apps_odoo_com/models/res_config.py (hunk 8 above)
Comment (fedoranvar) [reply apps_odoo_com/models/res_config.py]: check

File: apps_odoo_com/xmlrpc.py [hunk 9]
```diff
@@ -1,28 +1,30 @@
...
-    common = xmlrpclib.ServerProxy('{}/xmlrpc/2/common'.format(url))
-    models = xmlrpclib.ServerProxy('{}/xmlrpc/2/object'.format(url))
+    common = xmlrpc.client.ServerProxy('{}/xmlrpc/2/common'.format(url))
+    models = xmlrpc.client.ServerProxy('{}/xmlrpc/2/object'.format(url))
     uid = common.authenticate(dbname, username, password, {})
     if not uid:
         # for security reason log password only in debug level
         _logger.debug('Authenticate failed: %s', (dbname, username, password))
-        raise UserError(_('Authentication failed: %s'), (dbname, username,))
+        raise UserError(_('Authentication failed'))
```
Comment (KolushovAlexandr) [inline apps_odoo_com/xmlrpc.py]: why in the error message all useful information was cut off?

**Review by KolushovAlexandr — CHANGES_REQUESTED** [review]
This is not done:
'there are some default values in the res.config model fields it's ' '. Get rid of them'

How can I test demo data? It seems to be partially outdated for 12th version

File: apps_odoo_com/xmlrpc.py (hunk 9 above)
Comment (fedoranvar) [reply apps_odoo_com/xmlrpc.py]: there was an error

now i've returned

**Review by KolushovAlexandr — APPROVED** [review]

**Comment by yelizariev** [comment]
Будущее модуля туманно. Закрываю пока
